File: fszmq.props/fszmq.props.cs/Generation.cs
namespace fszmq.props
{
  using NUnit.Framework;
  using FsCheck;
  using System;
  using System.Linq;

  using CheckProperty = FsCheck.NUnit.PropertyAttribute;

  using fszmq;

  [TestFixture]
  public class Generation
  {
    [Test]
    public void PrintVersionDistribution ()
    {
      foreach (var pair in Distributions.Versions(25))
      {
        var version = pair.Key;
        var count   = pair.Value;
        Console.WriteLine("{0:00},{1}", count, version);
      }
    }

    [CheckProperty]
    public Boolean EncodeDecodeAreDuals_Failing (Byte[] data)
    {
      var processed = Z85.Decode(Z85.Encode(data));
      return processed.SequenceEqual(data);
    }

    [CheckProperty(Arbitrary = new []{ typeof(Generators) })]
    public Boolean EncodeDecodeAreDuals_Passing (Mod4Binary input)
    {
      var processed = Z85.Decode(Z85.Encode(input.Data));
      return processed.SequenceEqual(input.Data);
    }
  }
}

File: fszmq.props/fszmq.props.vb/Generation.vb
Imports CheckProperty = FsCheck.NUnit.PropertyAttribute

Namespace fszmq.props

  <TestFixture()> _
  Public Class Generation

    <Test()>
    Public Sub PrintVersionDistribution ()
      For Each pair in Distributions.Versions(25)
        Dim version = pair.Key
        Dim count   = pair.Value
        Console.WriteLine("{0:00},{1}", count, version)
      Next
    End Sub

    <CheckProperty()>
    Public Function EncodeDecodeAreDuals_Failing (ByVal data() As Byte) As Boolean
      Dim processed = Z85.Decode(Z85.Encode(data))
      Return processed.SequenceEqual(data)
    End Function

     <CheckProperty(Arbitrary := { GetType(Generators) })>
    Public Function EncodeDecodeAreDuals_Passing (ByVal input As Mod4Binary) As Boolean
      Dim processed = Z85.Decode(Z85.Encode(input.Data))
      Return processed.SequenceEqual(input.Data)
    End Function

  End Class

End Namespace
